blockly-ws-search/workspace_search.js

```diff
@@ -395,4 +401,75 @@ class WorkspaceSearch {
       Blockly.utils.dom.removeClass(blockPath, 'searchHighlight');
     });
   }
+
+  /**
+   * Scrolls workspace to bring given block into view.
+   * @param {Blockly.BlockSvg} block Block to bring into view.
+   * @private
+   */
+  scrollToVisible_(block) {
+    if (!this.workspace_.isMovable()) {
+      console.warn('Cannot scroll to block in a non-movable' +
+          'workspace.');
+      return;
+    }
+    // XY is in workspace coordinates.
+    const xy = block.getRelativeToSurfaceXY();
+    // Height/width is in workspace units.
+    const blockWidth = block.width;
+    const blockHeight = block.height;
+
+    const blockTop = xy.y;
+    const blockBottom = xy.y + blockHeight;
+    // In RTL the block's position is the top right of the block, not top left.
+    const blockLeft = this.workspace_.RTL ? xy.x - blockWidth: xy.x;
+    const blockRight = this.workspace_.RTL ? xy.x : xy.x + blockWidth;
+    // Workspace scale, used to convert from workspace coordinates to pixels.
+    const scale = this.workspace_.scale;
+
+    // Block bounds in pixels relative to the workspace origin (0,0 is centre).
+    const top = blockTop * scale;
+    const bottom = blockBottom * scale;
+    const left = blockLeft * scale;
+    const right = blockRight * scale;
+    const width = blockWidth * scale;
+    const height = blockHeight * scale;
```

Comment:
I would maybe consider combining these and then you could just have one set of constants and have bock in the name.

Reply:
Cleaned it up. It looks nicer now.